File: back/model/state.go
package model

// State holds public information about the state of a player
type State struct {
	MapImg      string `json:"mapImg"`
	AvatarImg   string `json:"avatarImg"`
	Finished    bool   `json:"finished"`
	Title       string `json:"title"`
	Text        string `json:"text"`
	MapPosition struct {
		X int `json:"x"`
		Y int `json:"y"`
	} `json:"mapPosition"`
	Template string `json:"template"`
	Tests    []Test `json:"tests"`
	Score    int    `json:"score"`
}

// InternalState holds private information about the state of a player
type InternalState struct {
	Finished bool `json:"finished"`
	Step     int  `json:"step"`
	Score    int  `json:"score"`
}

File: server/src/main/kotlin/com/zenika/nckotlinserver/model/State.kt
package com.zenika.nckotlinserver.model

/**
 * Holds public information about the state of a player
 */
data class State(
    val mapImg: String,
    val avatarImg: String,
    val finished: Boolean,
    val title: String,
    val text: String,
    val mapPosition: Coord,
    val template: String,
    val tests: List<Test>,
    val score: Int
)

/**
 * Holds private information about the state of a player
 */
data class InternalState(
    val playerId: String,
    val finished: Boolean = false,
    val step: Int = 0,
    val score: Int = 0
) : Entity {
    override fun id() = playerId
}